### PR TITLE
[ui][new] Adding new concur test

### PR DESCRIPTION
**Pull request by mcada**
//test: @concur

### CODE REVIEW

**Comment by syndesisqe** [comment]
Build finished. Tests PASSED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by mcada** [comment]
@avano only sf validation failed, test passed

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+@Slf4j
+public class ConcurValidationSteps {
+    private static String ACCESS_TOKEN = null;
+    private static final String ACCESS_TOKEN_ENDPOINT = "https://us-impl.api.concursolutions.com/oauth2/v0/token";
+    // you can find which endpoint you need and how to use it here:
+    // https://developer.concur.com/api-reference/expense/expense-report/reports.html
+    private static final String CONCUR_ENDPOINT_EXPENSE_REPORTS = "https://implementation.concursolutions.com/api/v3.0/expense/reports";
+
+    @When("^check that concur list or whatever we need to test is here \"([^\"]*)\"$")
+    public void invokeRequest(String something) {
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: not used?

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.fail;
+
+/**
+ * Maybe this will become handy in the future when we receive correct credentials for both
+ * OAuth and WS and both will point to the same instance. For now this class is not used.
+ */
+@Slf4j
+public class ConcurValidationSteps {
+    private static String ACCESS_TOKEN = null;
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: lower case

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+        String content = "{\"name\": \"report-from-testsuite\"}";
+
+        Response res = HttpUtils.doPostRequest(
+                CONCUR_ENDPOINT_EXPENSE_REPORTS,
+                content,
+                "application/json",
+                Headers.of("Authorization", "Bearer " + getAccessToken(), "Accept", "application/json")
+        );
+        log.info("Response code: " + res.code());
+        log.info(res.toString());
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: didn't you want `res.body().string()` ?

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+            return ACCESS_TOKEN;
+        }
+
+        log.info("Generating access token from concur API....");
+
+        FormBody.Builder formBuilder = new FormBody.Builder()
+                .add("grant_type", "password")
+                .add("credtype", "password");
+
+        //TODO: this needs to be loaded from credentials.json via accounts
```
Comment (avano) [inline ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: why is this todo?

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java (hunk 1 above)
Comment (mcada) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: Correct, this whole class is not used, you can see why in class javadoc. I don't want to remove it as it works and in the future we may receive both WS and OAuth credentials pointing to the same instance.

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java (hunk 4 above)
Comment (mcada) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: When I found out that I can not use those validation steps due to concur giving us two different credentials pointing to two different instances of concur, I stopped development of this class. But it works and may be used in the future.

File: ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java (hunk 1 above)
Comment (avano) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: imo commiting something that is not used anywhere is not a good practice

Comment (mcada) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: Well, I also thought about not including this verification file and just leave it in my branch if required later... Will do.

Comment (avano) [reply ui-tests/src/test/java/io/syndesis/qe/steps/other/ConcurValidationSteps.java]: looks better :dancer:

**Comment by mcada** [comment]
I wonder how that happened! I only dropped old concur commit and cherry-picked updated one :/ fixed...

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Comment by syndesisqe** [comment]
Build finished. Tests FAILED

**Pull request merged**